AsyncTwoPhaseIndexerTests.testFiveRuns failure

Build link: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+EAR/59/console

Reproduce:

```
./gradlew :x-pack:plugin:core:test --tests "org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexerTests.testFiveRuns" -Dtests.seed=127ACB67FDFE77CC -Dtests.security.manager=true -Dtests.locale=yue-Hant-HK -Dtests.timezone=Pacific/Gambier -Dcompiler.java=12 -Druntime.java=11
```

Failure:

```
org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexerTests > testFiveRuns FAILED
    java.lang.AssertionError: 
    Expected: <INDEXING>
         but: was <STARTED>
        at __randomizedtesting.SeedInfo.seed([127ACB67FDFE77CC:946DDD99B9859D60]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexerTests.testFiveRuns(AsyncTwoPhaseIndexerTests.java:396)
```

Could not reproduce locally

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
@@ -393,7 +393,6 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
             indexer.start();
             assertThat(indexer.getState(), equalTo(IndexerState.STARTED));
             assertTrue(indexer.maybeTriggerAsyncJob(System.currentTimeMillis()));
-            assertThat(indexer.getState(), equalTo(IndexerState.INDEXING));
             assertTrue(awaitBusy(() -> isFinished.get()));
             indexer.assertCounters();
         } finally {

